packages/core/monaco-editor/src/composables/useMonacoEditor.ts

```diff
@@ -0,0 +1,257 @@
+import { onActivated, onBeforeUnmount, onMounted, reactive, ref, toValue, watch } from 'vue'
+import { DEFAULT_MONACO_OPTIONS } from '../constants'
+import { unrefElement, useDebounceFn } from '@vueuse/core'
+
+// Monaco Editor Workers
+import EditorWorker from 'monaco-editor/esm/vs/editor/editor.worker?worker'
+import JsonWorker from 'monaco-editor/esm/vs/language/json/json.worker?worker'
+import CssWorker from 'monaco-editor/esm/vs/language/css/css.worker?worker'
+import HtmlWorker from 'monaco-editor/esm/vs/language/html/html.worker?worker'
+import TsWorker from 'monaco-editor/esm/vs/language/typescript/ts.worker?worker'
+
+import type * as monacoType from 'monaco-editor'
+import type { MaybeComputedElementRef, MaybeElement } from '@vueuse/core'
+import type { MonacoEditorStates, UseMonacoEditorOptions } from '../types'
+
+// singletons
+/** The Monaco instance once loaded */
+let monacoInstance: typeof monacoType | undefined
+/** Promise that resolves when Monaco is initialized */
+let monacoInitPromise: Promise<typeof monacoType> | null = null
+
+// cache
+const langCache = new Map<string, boolean>()
+
+/**
+ * Lazily load Monaco and configure workers only once.
+ */
+async function loadMonaco(language?: string): Promise<typeof monacoType> {
+  if (monacoInstance) {
+    if (language) {
+      monacoInstance.languages.register({ id: language })
+    }
+
+    return monacoInstance
+  }
+  if (monacoInitPromise && language && langCache.get(language)) {
+    return monacoInitPromise
+  }
+
+  monacoInitPromise = (async () => {
+    langCache.set(language || 'plaintext', true)
+
+    // Dynamically import the Monaco Editor core API
+    const monaco = await import('monaco-editor')
```

Comment:
Do you think we can import `monaco-editor` statically now?

Previously, we had to dynamically import `monaco-editor` in **konnect** because all composables there were loaded on every page, and we needed to avoid eagerly loading Monaco on pages that didn’t use it.

But since this composable is now in **public-ui-components**, maybe that concern no longer applies?

If we can switch to a static import, the benefit is that both `loadMonaco` and `init` would no longer need to be `async` functions.